### PR TITLE
Upgrade rust crates

### PR DESCRIPTION
**Pull request by piscisaureus**

### CODE REVIEW

File: build_extra/rust/run.py [hunk 1]
```diff
@@ -26,9 +27,19 @@
 env["GN_OUT_DIR"] = os.path.abspath(".")
 assert os.path.isdir(env["GN_OUT_DIR"])
 
+# Some crates (e.g. 'typenum') generate source files and place them in the
+# directory indicated by the 'OUT_DIR' environment variable, which is normally
+# set by Cargo. We pre-generate these files and store them in the source repo.
+# Therefore, set 'OUT_DIR' so these crates can find their generated sources.
```
Comment (ry) [inline build_extra/rust/run.py]: I was hoping this was general purpose support for build.rs files, but I guess not.

Comment (piscisaureus) [reply build_extra/rust/run.py]: > I was hoping this was general purpose support for build.rs files, but I guess not.

What do you mean? 
Build.rs files have been supported for the entire time; we just never captured any generated source files from the OUR_DIR.

Comment (ry) [reply build_extra/rust/run.py]: > Build.rs files have been supported for the entire time

Maybe I'm confused... is there an example third-party crate which executes its own build.rs ?

Comment (ry) [reply build_extra/rust/run.py]: The reason I'm asking is that build.rs support in rust.gni is my main hold up in importing [deno_typescript](https://github.com/ry/deno_typescript) into the main repo

Comment (piscisaureus) [reply build_extra/rust/run.py]: > Maybe I'm confused... is there an example third-party crate which executes its own build.rs ?

I don't understand what you mean maybe. Build.rs gets executed when I run gnargo. It doesn't participate in the ninja build.
There are dozens (probably >50) crates that use build.rs so please don't ask me to name all of them :)
'ring' is one example.

Comment (piscisaureus) [reply build_extra/rust/run.py]: Actually:

```
./registry/src/github.com-1ecc6299db9ec823/arrayvec-0.4.11/build.rs
./registry/src/github.com-1ecc6299db9ec823/bitflags-1.1.0/build.rs
./registry/src/github.com-1ecc6299db9ec823/byteorder-1.3.2/build.rs
./registry/src/github.com-1ecc6299db9ec823/httparse-1.3.4/build.rs
./registry/src/github.com-1ecc6299db9ec823/hyper-0.12.33/build.rs
./registry/src/github.com-1ecc6299db9ec823/kernel32-sys-0.2.2/build.rs
./registry/src/github.com-1ecc6299db9ec823/libc-0.2.62/build.rs
./registry/src/github.com-1ecc6299db9ec823/log-0.4.8/build.rs
./registry/src/github.com-1ecc6299db9ec823/memchr-2.2.1/build.rs
./registry/src/github.com-1ecc6299db9ec823/memoffset-0.5.1/build.rs
./registry/src/github.com-1ecc6299db9ec823/nix-0.14.1/build.rs
./registry/src/github.com-1ecc6299db9ec823/parking_lot_core-0.4.0/build.rs
./registry/src/github.com-1ecc6299db9ec823/proc-macro2-1.0.1/build.rs
./registry/src/github.com-1ecc6299db9ec823/rand-0.6.5/build.rs
./registry/src/github.com-1ecc6299db9ec823/rand_chacha-0.1.1/build.rs
./registry/src/github.com-1ecc6299db9ec823/rand_pcg-0.1.2/build.rs
./registry/src/github.com-1ecc6299db9ec823/ring-0.14.6/build.rs
./registry/src/github.com-1ecc6299db9ec823/ryu-1.0.0/build.rs
./registry/src/github.com-1ecc6299db9ec823/serde-1.0.99/build.rs
./registry/src/github.com-1ecc6299db9ec823/syn-1.0.3/build.rs
./registry/src/github.com-1ecc6299db9ec823/winapi-0.3.7/build.rs
./registry/src/github.com-1ecc6299db9ec823/ws2_32-sys-0.2.1/build.rs
```

Comment (ry) [reply build_extra/rust/run.py]: > Build.rs gets executed when I run gnargo

Ah ok - got it.

**Pull request merged**